src/services/users.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+class UsersService {
+  constructor() {
+    this.users = [];
+    this.lastUserId = 0;
+  }
+
+  resetData() {
+    this.users = [];
+    this.lastUserId = 0;
+  }
+
+  getAllUsers() {
+    return this.users;
+  }
+
+  getUserById(userId) {
+    return this.users.find(({ id }) => id === userId) || null;
+  }
+
+  addUser(userName) {
+    this.lastUserId++;
+
+    const newUser = {
+      name: userName,
+      id: this.lastUserId,
+    };
+
+    this.users.push(newUser);
+
+    return newUser;
+  }
+
+  removeUser(userId) {
+    this.users = this.users.filter(({ id }) => id !== +userId);
```

Comment:
For consistency, make the conversion on the controller level, just as you do everywhere else.
```suggestion
    this.users = this.users.filter(({ id }) => id !== userId);
```